### PR TITLE
add error state when !isNumber to Numeric component

### PR DESCRIPTION
**Pull request by VeraZab**
closes: #822 

screen shot 2019-01-09 at 11 15 41 am

same style as in this pr:
https://github.com/plotly/react-chart-editor/pull/811#issuecomment-452454940

### CODE REVIEW

**Comment by VeraZab** [comment]
Because NumericInput is the base widget used in all numerics (DualNumeric, derived components) adding validation to it should be working for anywhere there's numerics involved.

@dmt0 @nicolaskruchten ready for review

File: src/styles/components/widgets/_numeric-input.scss [hunk 1]
```diff
@@ -49,6 +49,26 @@
...
+  text-align: left;
+  border-radius: var(--border-radius-small);
+  padding: var(--spacing-quarter-unit) var(--spacing-quarter-unit) var(--spacing-quarter-unit)
+    var(--spacing-half-unit);
+  width: 62px;
+  vertical-align: middle;
+  font-size: inherit;
+  color: inherit;
+  font-family: inherit;
+}
```
Comment (dmt0) [inline src/styles/components/widgets/_numeric-input.scss]: We don't need to copy the whole class over, do we? We can just add a class with the difference.

File: src/components/widgets/NumericInput.js [hunk 2]
```diff
@@ -12,13 +12,25 @@ export default class NumericInput extends Component {
   constructor(props) {
     super(props);
 
-    this.state = {value: props.value};
+    this.state = {
+      value: props.value,
```
Comment (dmt0) [inline src/components/widgets/NumericInput.js]: I know this is not a new code addition, but assigning props to state is such an antipattern.

File: src/styles/components/widgets/_numeric-input.scss (hunk 1 above)
Comment (VeraZab) [reply src/styles/components/widgets/_numeric-input.scss]: yeah, I could do `@extend .numeric-input__number;`

File: src/components/widgets/NumericInput.js (hunk 2 above)
Comment (VeraZab) [reply src/components/widgets/NumericInput.js]: that's our way of not having plotly.js reject the value and override with its default

**Comment by nicolaskruchten** [comment]
On my screen the little gray fuzzy focus style hides the red... 

image

**Comment by VeraZab** [comment]
focussed version:
screen shot 2019-01-09 at 2 17 31 pm

**Comment by VeraZab** [comment]
Can I merge this one?

**Comment by nicolaskruchten** [comment]
same comments as the other validation one: there's no easy "back out to defaults" mode... once it's red it stays red and deleting it doesn't undo... Can we have it accept `""` such that plotly.js will then go back to defaults?

**Comment by nicolaskruchten** [comment]
:dancer:

**Pull request merged**